ipfs/bitswap/stream.nim

```diff
@@ -0,0 +1,38 @@
+import pkg/chronos
+import pkg/protobuf_serialization
+import pkg/libp2p/stream/connection
+import ./messages
+
+export messages
+
+const MaxMessageSize = 8 * 1024 * 1024
+
+type
+  BitswapStream* = ref object
```

Comment:
I'm not sure about the need of this type. I think we can get by without it and instead add an RPC handler similar to how its done in `PubSub` - https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/floodsub.nim#L83-L128. 

It's probably also usefull to addopt a similar structure to pubsub in general, with a "bitswap peer" type that handles each peer individually and has it's own read/write mechanism similar to https://github.com/status-im/nim-libp2p/blob/master/libp2p/protocols/pubsub/pubsubpeer.nim. Note how the peer handles its own connection and has a read loop that parses messages and then invokes the main protocol handler.